StPicoDstMaker/StPicoDstMaker.cxx

```diff
@@ -561,7 +561,7 @@ void StPicoDstMaker::fillTracks()
   {
     StMuTrack* pTrk = (StMuTrack*)mMuDst->primaryTracks(i);
     if (!pTrk) continue;
-    if (pTrk->id() < 0 || pTrk->id() >= 50000)
+    if (pTrk->id() < 0)
```

Comment:
Ok, I looked into this again more carefully. I did overlook the fact that we are talking about a fixed size c-array...  But anyway my main point remains the same 1) muDst can return any number of tracks consistent with the short type. This is not an implementation detail of muDst but its interface. 2) picoDst should not reject tracks based on an assumed implementation of short and that's what the check was doing.
There are other problems with this lookup map they create. It shouldn't even be a class member and it would make more sense to make its size dynamic so we don't worry about what muDst returns.